### PR TITLE
Updated Gradle info

### LINKED ISSUE #9234: Add documentation for common changes to generated gradle files
### What information is missing?

Example: One of the [most liked flutter-related questions on stack overflow](https://stackoverflow.com/questions/52060516/flutter-how-to-change-android-minsdkversion-in-flutter-project) asks how to change the minSdkVersion in a Flutter app for android. Note that while the top answer here is correct, it doesn't cover how plugin authors should change the minSdkVersion (to be fair the question doesn't ask that, but the post will be the top result on google for a plugin author who searches about changing the minSdkVersion in their plugin). Also, some of the other (highly liked) answers have different suggestions that aren't ideal.

Another one of the most liked: [changing your application id](https://stackoverflow.com/questions/51534616/how-to-change-package-name-in-flutter). This even has a TODO: in the [template for the module level build.gradle](https://github.com/flutter/flutter/blob/b0f90db6e1e66441039da8ebb474a137d4ab0fbe/packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl#L36C58-L36C120).

### How would you like us to fix this problem?

We should add a page covering these common cases when a developer will need to interact with their gradle files, clarifying the steps they need to take and providing examples (maybe also covering changes to the android manifest in the same page).

### PR DESCRIPTION
**Pull request by atsansone**
Fixes #9234

### CODE REVIEW

**Comment by domesticmouse** [comment]
Can you please fix the conflicts?

File: src/content/deployment/android.md [hunk 1]
```diff
@@ -92,8 +92,7 @@ To find out the latest version, visit [Google Maven][].
 +<style name="NormalTheme" parent="Theme.MaterialComponents.DayNight.NoActionBar">
 ```
 
-<a id="signing-the-app"></a>
-## Sign the app
+## Sign the app {:#signing-the-app}
```
Comment (parlough) [inline src/content/deployment/android.md:95]: Please keep the separate anchor here so they both continue to work. There's old links to both headers from outside the site.

Comment (atsansone) [reply src/content/deployment/android.md:95]: In the site checks out as OK.

Comment (parlough) [reply src/content/deployment/android.md:95]: Yes in the site is fine, but these extra anchors are here for outside the side, such as from old framework and tool versions, other sites, etc.

File: src/content/deployment/android.md [hunk 2]
```diff
@@ -296,66 +302,81 @@ check out the official [Android documentation][multidex-docs].
...
+| `application`        | Edit the `android:label` in the [`application`][applicationtag] tag to reflect the final name of the app. |
+| `uses-permission`    | Add the `android.permission.INTERNET` [permission][permissiontag] value to the `android:name` attribute if your app needs Internet access. The standard template doesn't include this tag but allows Internet access during development to enable communication between Flutter tools and a running app. |
+
+{:.table .table-striped}
+
+## Review or change the Gradle build configuration {:#review-the-gradle-build-configuration}
+
+To verify the Android build configuration,
+review the `android` block in the default
+[Gradle build file][gradlebuild].
```
Comment (reidbaker) [inline src/content/deployment/android.md:331]: There can be multiple build.gradle files and it is hard to know which one is default. 
This code is normally in the android/app/build.gradle file

File: src/content/deployment/android.md [hunk 3]
```diff
@@ -296,66 +302,81 @@ check out the official [Android documentation][multidex-docs].
...
 
-`applicationId`
-: Specify the final, unique [application ID][].
-  
-`minSdk`
-: Specify the [minimum API level][] on which you designed the app to run.
-  Defaults to `flutter.minSdkVersion`.
+| Property             | Purpose                                                                                                                                                                                                            | Default Value               |
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------|
+| `compileSdk`         | The Android API level with which your app should be compiled.                                                                                                                                                      | `flutter.compileSdkVersion` |
```
Comment (reidbaker) [inline src/content/deployment/android.md:368]:
```suggestion
| `compileSdk`         | The Android API level with which your app should be compiled.  This should normally be the highest version available.         
```

File: src/content/deployment/android.md [hunk 4]
```diff
@@ -296,66 +302,81 @@ check out the official [Android documentation][multidex-docs].
...
+| `defaultConfig`      |  |  |
+| `.applicationId`     | The final, unique [application ID][] that identifies your app.                                                                                                                                                     |                             |
+| `.minSdk`            | The [minimum Android API level][] for which you designed the app to run.                                                                                                                                           | `flutter.minSdkVersion`     |
+| `.targetSdk`         | The target Android API level for which you designed the app to run.                                                                                                                                                | `flutter.targetSdkVersion`  |
```
Comment (reidbaker) [inline src/content/deployment/android.md:372]: https://stackoverflow.com/questions/26694108/what-is-the-difference-between-compilesdkversion-and-targetsdkversion
This stackoverflow question does a good job explaining the difference

**Review by reidbaker — APPROVED** [review]

**Pull request merged**